imports/plugins/core/product/server/no-meteor/schemas/schema.graphql

```diff
@@ -6,3 +6,68 @@ type ProductConfiguration {
   "The ProductVariant ID"
   productVariantId: ID!
 }
+
+"Operation"
+type Operation {
+  "Id of document"
+  _id: String
+}
+
+"A bulk write error type"
+type WriteError {
+  "Error code"
+  code: Int
+
+  "Error message"
+  errmsg: String
+
+  "Index of of error"
+  index: Int
+
+  "Failed operation info"
+  op: Operation
+}
+
+"Input for adding tags to products in bulk"
+input ProductsBulkOperationInput {
+  "An optional string identifying the mutation call, which will be returned in the response payload"
+  clientMutationId: String
+
+  "An array of product ids to which an array of tags will be added"
+  productIds: [ID]
+
+  "An array of tag ids to add to an array of products"
+  tagIds: [ID]
+}
+
+"Response payload for `bulkAddTagsToProducts` mutation"
+type ProductsBulkOperationPayload  {
+  "The same string you sent with the mutation params, for matching mutation calls with their responses"
+  clientMutationId: String
+
+  "The number of products found"
+  foundCount: Int
+
+  "The number of products for which a match was not found"
+  notFoundCount: Int
+
+  "The number of products successfully updated"
+  updatedCount: Int
+
+  "An array of write errors if any were generated"
+  writeErrors: [WriteError]
```

Comment:
Overall this payload strikes me as being not too useful for the client. I think maybe you should parse the result a bit and convert it into something like this for the response payload:

```js
{
  results: [
    { documentId: "productId1", wasFound: true, wasUpdated: true, errorMessage: null },
    { documentId: "productId2", wasFound: true, wasUpdated: false, errorMessage: "something went wrong" } 
  ]
}
```

This would make it easiest for browser code to display the information in helpful way to the user.

Reply:
It would be ideas to have the details you mention above for each `documentId` updated. However, the `bulkWrite` operation does not return such information for for each updated document, it only returns those details if there was a write error. https://docs.mongodb.com/manual/reference/method/db.collection.bulkWrite/index.html